app/src/androidTest/java/com/conupods/Settings/SettingsActivityTest.java

```diff
@@ -0,0 +1,85 @@
+package com.conupods.Settings;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.view.View;
+import android.widget.CheckBox;
+import androidx.test.rule.ActivityTestRule;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.conupods.R;
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+
+import static org.junit.Assert.*;
+
+public class SettingsActivityTest {
+
+    @Rule
+    public ActivityTestRule<SettingsActivity> mActivityTestRule = new ActivityTestRule<>(SettingsActivity.class);
+    private SettingsActivity mActivity = null;
+
+    @Rule
+    public ActivityTestRule<SettingsInfoActivity> mActivityInfoTestRule = new ActivityTestRule<>(SettingsInfoActivity.class);
+    private SettingsInfoActivity mActivityInfo = null;
+
+    @Rule
+    public ActivityTestRule<SettingsPersonalActivity> mActivityPersonalTestRule = new ActivityTestRule<>(SettingsPersonalActivity.class);
+    private SettingsPersonalActivity mActivityPersonal = null;
+
+    @Before
+    public void setUp() {
+        mActivity = mActivityTestRule.getActivity();
+        mActivityInfo = mActivityInfoTestRule.getActivity();
```

Comment:
See my comment on how to change this file that I sent you over discord as you are doing redundant initializations for each test and there are tests failing on my machine that should be looked at :
* `goToSettingsPersonalActivityFromSettingsActivityTest`
* `changePreferencesTest`